src/api/app/assets/stylesheets/webui/comments.scss

```diff
@@ -21,6 +21,49 @@
   }
 }
 
+.comment-bubble {
+  $comment-bubble-outline: $gray-400;
+  @extend .rounded;
+  @extend .pt-3;
+  @extend .px-3;
+  @extend .mb-2;
+  @extend .d-inline-flex;
+  border: 1px solid $comment-bubble-outline;
+  position: relative;
+
+  // Arrow on the top left of the comment bubble
+  $comment-bubble-arrow-top-position: 0;
+  $comment-bubble-arrow-left-position: 16px;
+  $comment-bubble-arrow-size: 16px;
+  // This creates two triangles.
+  // `::before` creates a triangle which matches the color of the comment bubble's outline.
+  //
+  // `::after` creates a triangle which matches the color of Bootstrap's card background color.
+  // It is displayed on top of the one created by `::before` with a small offset.
+  //
+  // Together, they produce a single comment bubble arrow... Magic!
```

Comment:
I've got inspired by some of the CSS [here](https://sharkcoder.com/visual/shapes#section4) and [there](https://sharkcoder.com/visual/shapes#section12). I use Sass variables in order to describe what those values are, hopefully making this easier to understand.